cfgmgr/buffermgr.cpp

```diff
@@ -185,9 +195,26 @@ task_process_status BufferMgr::doSpeedUpdateTask(string port, bool admin_up, con
                     }
                 }
             }
-
-            continue;
         }
+        // Clear applied recorded
+        if (m_portLosslessPgs.count(port) != 0)
+        {
+            m_portLosslessPgs.erase(port);
+        };
+        // Clear recorded pg for admin_down port
+        if (m_portPgLookup.count(port) != 0)
+        {
+            m_portPgLookup.erase(port);
+        }
+        return task_process_status::task_success;
+    }
+
+    for (auto lossless_pg : lossless_pgs)
+    {
+        vector<FieldValueTuple> fvVectorPg, fvVectorProfile;
+        string buffer_pg_key = port + m_cfgBufferPgTable.getTableNameSeparator() + lossless_pg;
+
+        m_cfgBufferPgTable.get(buffer_pg_key, fvVectorPg);
```

Comment:
the buffer profile creation/existence check is not dependent on the pgs. so that needn26't be in a loop.  This entire block should move to after L258 and the for loop should include everything from L261.

The logic is create the buffer profile if does not exist. Then go through each lossless pg and check the buffer profile it references. if it is the same, don't do anything otherwise apply the buffer profile to the pg

Reply:
Updated. Thanks for the great suggestion.